packages/extension/src/common/keyNavigationService.js

```diff
@@ -17,88 +16,48 @@ export const validKeys = {
   esc: 27,
 };
 
-function getNewPostEl(keyCode, currentElement, insaneMode) {
-  if (keyCode === validKeys.h && !insaneMode) return getAbovePost(currentElement);
+export const validKeysValues = Object.values(validKeys);
 
-  if (keyCode === validKeys.l && !insaneMode) return getBelowPost(currentElement);
+function navigateFeed(keyCode, currentElement, insaneMode) {
+  if (keyCode === validKeys.h && !insaneMode) return getPreviousPost(currentElement);
 
-  if (keyCode === validKeys.j && insaneMode) return getBelowPost(currentElement);
+  if (keyCode === validKeys.l && !insaneMode) return getNextPost(currentElement);
 
-  if (keyCode === validKeys.k && insaneMode) return getAbovePost(currentElement);
+  if (keyCode === validKeys.j) return getBelowPost(currentElement, insaneMode);
 
-  return currentElement;
-}
-
-function triggerBookmark(post) {
-  if (post.ad) return null;
-
-  return post.$emit('bookmark', { post: post.post, bookmarked: !post.post.bookmarked });
-}
-
-function getCurrentPost(posts, current) {
-  if (!current) return null;
+  if (keyCode === validKeys.k) return getAbovePost(currentElement, insaneMode);
 
-  const postOrAdProp = current.post || current.ad;
-
-  return posts.find(article => [article.ad, article.post].indexOf(postOrAdProp) !== -1);
+  return currentElement;
 }
 
-function navigate(keyCode, posts, search, backToMainFeed, { current, insaneMode }) {
-  if (keyCode === validKeys.esc) return backToMainFeed();
+export function navigateDaily(keyCode, posts, hoveredPost, insaneMode) {
+  const foundElement = hoveredPost
+    ? navigateFeed(keyCode, hoveredPost.$el, insaneMode)
+    : getFirstPostOnFeed(insaneMode);
 
-  if (posts.length === 0) return null;
+  const post = posts.find(article => article.$el === foundElement);
 
-  const item = getCurrentPost(posts, current);
+  if (post) return post;
 
-  if (Object.values(validKeys).indexOf(keyCode) === -1) return null;
-
-  if (keyCode === validKeys['/']) return search.enable();
-
-  if (keyCode === validKeys.b && item) return triggerBookmark(item);
-
-  const element = !item
-    ? getTopLeftMostPostEl(posts, insaneMode)
-    : getNewPostEl(keyCode, item.$el, insaneMode);
-
-  const selectedPost = getPostByElement(posts, element);
-
-  if (selectedPost === item) return selectedPost;
-
-  hoverPost(selectedPost);
-
-  return selectedPost;
-}
+  const nextPostAfterEmptyAdElement = getNextPost(foundElement, insaneMode);
 
-function navigateDaily({ keyCode, target }) {
-  if (target instanceof HTMLInputElement || target instanceof HTMLTextAreaElement) {
-    return;
-  }
```

Comment:
Instead of manually turning on/off the keybindings. This code snippet can detect automatically if the keypress is targeted to an input element.
I think it's a better solution as it is future-proof.

Reply:
Ohhhh, that's amazing, I never knew this. Thank you for the info!

Reply:
You bet! I've implemented it a while ago 🤩

Reply:
Yeah, I saw the changes. Sincere apologies for omitting! I didn't realize it was for that.